packages/template-retail-react-app/app/hooks/use-auth-modal.js

```diff
@@ -173,15 +183,23 @@ export const AuthModal = ({
                     })
                 }
             },
-            email: async (data) => {
-                // Handle resend passwordless email logic here
+            email: async () => {
+                try {
+                    authorizePasswordlessLogin(passwordlessLoginEmail)
+                } catch (e) {
+                    form.setError('global', {
+                        type: 'manual',
+                        message: formatMessage(API_ERROR_MESSAGE)
+                    })
+                }
```

Comment:
This code looks a lot like the code in the Password type condition, under the sub-condition of Passwordless type. Why do we have this repetition? and it is possible not to have it?

Reply:
The reason we have this separate section `email` is to handle submit while on the `email` view. It resends the passwordless email which is why it sends the same `authorizePasswordlessLogin` call in this condition. 

I've extracted the logic into a helper function `handlePasswordlessLogin` so that at the least it is clear both conditions call the same function

<img width="1792" alt="Screenshot 2024-12-18 at 3 05 57 PM" src="https://github.com/user-attachments/assets/b9b37d80-abfe-4849-a810-7fcf68399950" />